Problem when used together with Phoenix library

### Platform

Android

### Installed

None

### Version

0.5.0

### Steps to Reproduce

1. Use new init function that does not take `Context` and relies on `ContentProvider`
2. In the app at some point do `ProcessPhoenix.triggerRebirth(context)`



### Expected Result

App should not crash

### Actual Result

App crashes with
```
Caused by: kotlin.UninitializedPropertyAccessException: lateinit property applicationContext has not been initialized
	at io.sentry.kotlin.multiplatform.SentryInit_androidKt.getApplicationContext(SentryInit.android.kt:17)
	at io.sentry.kotlin.multiplatform.SentryInit_androidKt.initSentry(SentryInit.android.kt:14)
	at io.sentry.kotlin.multiplatform.SentryBridge.init(SentryBridge.jvm.kt:21)
	at io.sentry.kotlin.multiplatform.Sentry.init(SentryKMP.kt:43)
```

--- a/CHANGELOG.md
+++ b/CHANGELOG.md
@@ -14,6 +14,10 @@ pod("Sentry") {
 }
 ```
 
+### Fixes
+
+- Don't crash app when `applicationContext` is not available ([#217](https://github.com/getsentry/sentry-kotlin-multiplatform/pull/217))
+
 ### Enhancements
 
 - Make `setSentryUnhandledExceptionHook` public ([#208](https://github.com/getsentry/sentry-kotlin-multiplatform/pull/208))

--- a/sentry-kotlin-multiplatform/src/androidMain/kotlin/io/sentry/kotlin/multiplatform/SentryInit.android.kt
+++ b/sentry-kotlin-multiplatform/src/androidMain/kotlin/io/sentry/kotlin/multiplatform/SentryInit.android.kt
@@ -5,16 +5,24 @@ import android.content.ContentValues
 import android.content.Context
 import android.database.Cursor
 import android.net.Uri
+import android.util.Log
 import io.sentry.android.core.SentryAndroid
 import io.sentry.kotlin.multiplatform.extensions.toAndroidSentryOptionsCallback
 
 internal actual fun initSentry(configuration: OptionsConfiguration) {
     val options = SentryOptions()
     configuration.invoke(options)
-    SentryAndroid.init(applicationContext, options.toAndroidSentryOptionsCallback())
-}
 
-internal lateinit var applicationContext: Context
+    val context = applicationContext ?: run {
+        // TODO: add logging later
+        return
+    }
+
+    SentryAndroid.init(context) { sentryOptions ->
+        options.toAndroidSentryOptionsCallback().invoke(sentryOptions)
+    }}
+
+internal var applicationContext: Context? = null
     private set
 
 public actual typealias Context = Context

--- a/sentry-kotlin-multiplatform/src/androidMain/kotlin/io/sentry/kotlin/multiplatform/SentryInit.android.kt
+++ b/sentry-kotlin-multiplatform/src/androidMain/kotlin/io/sentry/kotlin/multiplatform/SentryInit.android.kt
@@ -5,7 +5,6 @@ import android.content.ContentValues
 import android.content.Context
 import android.database.Cursor
 import android.net.Uri
-import android.util.Log
 import io.sentry.android.core.SentryAndroid
 import io.sentry.kotlin.multiplatform.extensions.toAndroidSentryOptionsCallback
 
@@ -20,7 +19,8 @@ internal actual fun initSentry(configuration: OptionsConfiguration) {
 
     SentryAndroid.init(context) { sentryOptions ->
         options.toAndroidSentryOptionsCallback().invoke(sentryOptions)
-    }}
+    }
+}
 
 internal var applicationContext: Context? = null
     private set

--- a/sentry-kotlin-multiplatform/src/androidUnitTest/kotlin/io/sentry/kotlin/multiplatform/SentryContextProviderTest.kt
+++ b/sentry-kotlin-multiplatform/src/androidUnitTest/kotlin/io/sentry/kotlin/multiplatform/SentryContextProviderTest.kt
@@ -7,6 +7,7 @@ import org.junit.Before
 import org.junit.Test
 import org.junit.runner.RunWith
 import org.mockito.Mockito.mock
+import kotlin.test.assertNotNull
 
 @RunWith(AndroidJUnit4::class)
 class SentryContextProviderTest : BaseSentryTest() {
@@ -22,8 +23,8 @@ class SentryContextProviderTest : BaseSentryTest() {
     class SentryContextOnCreateTest : BaseSentryTest() {
         @Test
         fun `onCreate initializes applicationContext`() {
-            // Simple call to the lateinit applicationContext to make sure it's initialized
-            applicationContext
+            // Simple call to the applicationContext to make sure it's initialized
+            assertNotNull(applicationContext)
         }
     }
 

